.github/workflows/flutter_android_test.yml

```diff
@@ -0,0 +1,121 @@
+on:
+  workflow_call:
+    inputs:
+      flutter_version:
+        required: false
+        type: string
+        default: '3.3.1'
+      rust_version:
+        required: false
+        type: string
+        default: '1.64.0'
+      device:
+        required: true
+        type: string
+      uses_cache:
+        required: false
+        type: boolean
+        default: true
+      timeout_minutes:
+        required: false
+        type: number
+        default: 20
+
+jobs:
+  flutter_android_test:
+    runs-on: macos-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # #499, https://github.com/actions/virtual-environments/issues/5595
+      - name: Configure ndk
+        run: |
+            ANDROID_HOME=$HOME/Library/Android/sdk
+            SDKMANAGER=$ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager
+
+            echo y | $SDKMANAGER "ndk;21.4.7075529"
+
+            ln -sfn $ANDROID_HOME/ndk/21.4.7075529 $ANDROID_HOME/ndk-bundle
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ inputs.rust_version }}
+          override: true
+
+      - uses: subosito/flutter-action@v2
+        with:
+          channel: "stable"
+          flutter-version: ${{ inputs.flutter_version }}
+          architecture: x64
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: "8.x" # "betterprogramming.pub" says must be java "8"
+
+      - uses: actions/cache@v3
+        if: ${{ inputs.uses_cache }}
+        id: cache-deps
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ./frb_example/with_flutter/rust/target
+          key: ${{ runner.os }}-cargo-integrate-android-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Instead of hard-coding the Rust version, we can also add it to the cache key as well and let Cargo follow the stable channel. You can use `actions-rs/toolchain`'s output [`rustc_hash`](https://github.com/actions-rs/toolchain#outputs) to achieve caching per Rust version.

Reply:
done in d285644a1246dcb2cbd51c49c0b17cf2a3cb810b